pythran/pythonic/numpy/ndarray/reshape.hpp

```diff
@@ -27,7 +28,20 @@ namespace numpy
                             types::ndarray<T, NpS>>::type
     reshape(types::ndarray<T, pS> const &expr, NpS const &new_shape)
     {
-      long where = sutils::find(new_shape, -1);
+      long where = -1;
```

Comment:
I understand why you're doing that, but would that be possible to extend `sutils::find`` so that it accepts an extra `start` argument. In that case you could just chain two calls to `sutils::find` to detect errors... You could also make `sutils::find` accept a lambda predicate instead of a value to check for any negative value. I want to avoid creating a temporary array like you do, that's not super efficient.

Reply:
Yes, I don't like it either, I like your implementation of sutils::find and I was thinking about extending it to test for < ... I'll try your ideas.